### PR TITLE
Add indeterminate checkbox option to kits

### PR DESCRIPTION
**Pull request by coleerikson**
#### Screens
indeterminate

#### Breaking Changes

No - Adds props

#### Runway Ticket URL

https://nitro.powerhrg.com/runway/backlog_items/NUXE-229

#### How to test this

Visual/CSS only. If prop is activated, then should render minus sign within checkbox ui.

#### Checklist:

- [ ] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **URGENCY** Please select a release milestone
- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [x] **SCREENSHOT** Please add a screen shot or two.
- [x] **SPECS** Please cover your changes with specs.

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_checkbox/_checkbox.jsx [hunk 1]
```diff
@@ -33,21 +34,29 @@ const Checkbox = (props: CheckboxProps) => {
...
   const classes = classnames(
-    buildCss('pb_checkbox_kit', { checked, error }),
+    buildCss('pb_checkbox_kit', { checked, error, indeterminate }),
     globalProps(props),
     className
   )
 
+  useEffect(() => {
+    if (checkRef.current) {
+      checkRef.current.indeterminate = indeterminate
```
Comment (coleerikson) [inline playbook/app/pb_kits/playbook/pb_checkbox/_checkbox.jsx:56]: Indeterminate as attribute for a traditional DOM elements needs to be added onmount or rerender. Hence the `useEffect` & `useRef`.

**Review by thestephenmarshall — COMMENTED** [review]
Nice work. One question: given the complexity of this new checkbox variant, do you think it would be possible to add more test coverage (attributes, etc)?

**Comment by coleerikson** [comment]
No, I don't mind. 👍

**Review by jasperfurniss — APPROVED** [review]

**Comment by coleerikson** [comment]
@jasperfurniss my last commit added some tests. Let me know if you think I could add other types.

**Review by thestephenmarshall — APPROVED** [review]

**Pull request merged**